backend/src/apiserver/client_manager.go

```diff
@@ -226,6 +226,11 @@ func initDBClient(initConnectionTimeout time.Duration) *storage.DB {
 		glog.Fatalf("Failed to drop unique key on experiment name. Error: %s", response.Error)
 	}
 
+	response = db.Model(&model.Pipeline{}).RemoveIndex("Name")
+	if response.Error != nil {
+		glog.Fatalf("Failed to drop unique key on pipeline name. Error: %s", response.Error)
+	}
+
```

Comment:
Should the migration code be wrapped in a transaction (remove index + add new)?
This is in order to not leave the database in an invalid state, in case the migration fails to complete.
Also, is this code idempotent? Meaning, what would happen if the "Name" index doesn't exist? Would it fail?

Maybe `gorm` does it automatically if the (name, namespace) index is declared on the Go struct (in AutoMigrate)?

Reply:
That's my understanding as well, and yes I also believe the operation is idempotent, I remember testing this locally. I added a comment here in how it would look like given KFP switches to the ```gorm``` main fork. 
https://github.com/kubeflow/pipelines/issues/5125 

I think it's best to leave the operations separately, that's the pattern currently being used with this fork. 

We tested this from Kubeflow 1.1 and we are using the code I wrote in production. Which has this logic, we did not observe any issues and we also tested it on the MySQL in cluster which Kubeflow comes with and an Azure MySQL.